### ISSUE TITLE
CoreOS on EC2 losing network connection once a day

### ISSUE THREAD

**Issue by canassa** [coreos/bugs#1551]
# Issue Report #

## Bug ##

I have a EC2 instance provisioned using the CoreOS Stable AMI with the following cloud-config:

```yaml
coreos:
  units:
    - name: esdata.mount
      command: start
      content: |
        [Mount]
        What=/dev/xvdf
        Where=/esdata
        Type=ext4

    - name: "kibana.service"
      command: "start"
      content: |
        [Unit]
        Description=Kibana
        After=elasticsearch.service
        Requires=elasticsearch.service

        [Service]
        TimeoutStartSec=0
        TimeoutStopSec=60
        Restart=always
        ExecStartPre=-/usr/bin/docker rm kibana
        ExecStartPre=/usr/bin/docker pull kibana:latest
        ExecStart=/usr/bin/docker run --rm -p 80:5601 --name kibana --link elasticsearch kibana:latest kibana
        ExecStop=/usr/bin/docker stop -t 60 kibana

        [Install]
        WantedBy=multi-user.target

    - name: "elasticsearch.service"
      command: "start"
      content: |
        [Unit]
        Description=Elasticsearch
        After=docker.service esdata.mount
        Requires=docker.service esdata.mount

        [Service]
        TimeoutStartSec=0
        TimeoutStopSec=60
        Restart=always
        ExecStartPre=-/usr/bin/docker rm elasticsearch
        ExecStartPre=/usr/bin/docker pull elasticsearch:latest
        ExecStart=/usr/bin/docker run --rm -p 9200:9200 -v /esdata:/usr/share/elasticsearch/data --name elasticsearch elasticsearch:latest elasticsearch -Des.network.host=0.0.0.0
        ExecStop=/usr/bin/docker stop -t 60 elasticsearch

        [Install]
        WantedBy=multi-user.target
```

The instance runs just fine, the problem is that every other day I lose network connection to it and I have to manually reboot it using the Amazon console.

Things I have already tried
 - Disabling IPV6
 - Destroying the machine and provisioning it again

I think these are logs from when the problem happened:

```
Aug 31 23:54:35 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]:         at java.lang.Thread.run(Thread.java:745)
Aug 31 23:54:35 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]:         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
Aug 31 23:54:34 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]:         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
Aug 31 23:54:33 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]:         at org.elasticsearch.transport.TransportService$TimeoutHandler.run(TransportService.java:679)
Aug 31 23:54:32 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: ReceiveTimeoutTransportException[[Jordan Seberius][172.17.0.2:9300][cluster:monitor/nodes/stats[n]] request_id [519619] timed out after [15273ms]]
Aug 31 23:54:32 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:53:34,202][DEBUG][action.admin.cluster.node.stats] [Jordan Seberius] failed to execute on node [pNJoJaxxT0ipYHlhA78ntw]
Aug 31 23:53:59 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCPv6 CLIENT: Next retransmission in 2min 10s
Aug 31 23:53:58 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCPv6 CLIENT: Sent SOLICIT
Aug 31 23:53:48 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:53:34,202][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66369][24560] duration [2.9s], collections [1]/[5.3s], total [2.9s]/[
Aug 31 23:53:31 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:53:27,851][WARN ][transport                ] [Jordan Seberius] Received response for a request that has timed out, sent [37407ms] ago, timed out
Aug 31 23:52:55 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:52:44,464][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66356][24559] duration [1.9s], collections [1]/[4.2s], total [1.9s]/[
Aug 31 23:52:42 ip-172-31-30-108.eu-central-1.compute.internal locksmithd[809]: Unlocking old locks failed: [etcd.service etcd2.service] are inactive. Retrying in 5m0s.
Aug 31 23:52:40 ip-172-31-30-108.eu-central-1.compute.internal locksmithd[809]: [etcd.service etcd2.service] are inactive
Aug 31 23:52:31 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:52:21,158][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66349][24558] duration [10.5s], collections [1]/[1.8s], total [10.5s]
Aug 31 23:51:56 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCPv6 CLIENT: Next retransmission in 2min 2s
Aug 31 23:51:55 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCPv6 CLIENT: Sent SOLICIT
Aug 31 23:51:58 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:51:51,999][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66344][24557] duration [7.7s], collections [1]/[11.1s], total [7.7s]/
Aug 31 23:51:47 ip-172-31-30-108.eu-central-1.compute.internal docker[9846]: {"type":"log","@timestamp":"2016-08-31T23:50:25+00:00","tags":["status","plugin:elasticsearch","error"],"pid":9,"name":"plugin:elasticsearch","state":"red","me
Aug 31 23:50:55 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:50:44,149][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66321][24556] duration [8.2s], collections [1]/[11.2s], total [8.2s]/
Aug 31 23:50:21 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:50:13,089][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66311][24555] duration [2.3s], collections [1]/[6.3s], total [2.3s]/[
Aug 31 23:50:18 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:50:07,208][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66310][24554] duration [2.9s], collections [1]/[5.3s], total [2.9s]/[
Aug 31 23:50:06 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCPv6 CLIENT: Next retransmission in 1min 49s
Aug 31 23:50:05 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCPv6 CLIENT: Sent SOLICIT
Aug 31 23:50:03 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:49:49,771][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66302][24553] duration [1.1s], collections [1]/[3.3s], total [1.1s]/[
Aug 31 23:49:31 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:49:29,452][INFO ][monitor.jvm              ] [Jordan Seberius] [gc][young][66293][24551] duration [884ms], collections [1]/[2.4s], total [884ms]
Aug 31 23:49:28 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:49:23,827][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66290][24550] duration [1.5s], collections [1]/[3.8s], total [1.5s]/[
Aug 31 23:49:21 ip-172-31-30-108.eu-central-1.compute.internal docker[9846]: {"type":"log","@timestamp":"2016-08-31T23:48:07+00:00","tags":["status","plugin:elasticsearch","error"],"pid":9,"name":"plugin:elasticsearch","state":"red","me
Aug 31 23:49:09 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:49:07,255][INFO ][monitor.jvm              ] [Jordan Seberius] [gc][young][66280][24549] duration [951ms], collections [1]/[2.1s], total [951ms]
Aug 31 23:49:02 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:48:56,279][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66275][24548] duration [1.6s], collections [1]/[3.6s], total [1.6s]/[
Aug 31 23:48:45 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:48:43,096][INFO ][monitor.jvm              ] [Jordan Seberius] [gc][young][66268][24547] duration [870ms], collections [1]/[1.6s], total [870ms]
Aug 31 23:48:28 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:48:27,485][INFO ][monitor.jvm              ] [Jordan Seberius] [gc][young][66258][24546] duration [927ms], collections [1]/[3.3s], total [927ms]
Aug 31 23:48:23 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:48:16,760][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66253][24545] duration [1.4s], collections [1]/[2.2s], total [1.4s]/[
Aug 31 23:48:12 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:48:10,620][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66250][24544] duration [2.8s], collections [1]/[5s], total [2.8s]/[3m
Aug 31 23:48:08 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCPv6 CLIENT: Next retransmission in 1min 57s
Aug 31 23:48:08 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCPv6 CLIENT: Sent SOLICIT
Aug 31 23:48:04 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:48:00,454][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66247][24543] duration [1.2s], collections [1]/[3s], total [1.2s]/[3m
Aug 31 23:47:24 ip-172-31-30-108.eu-central-1.compute.internal systemd-timesyncd[704]: Synchronized to time server 81.7.16.52:123 (0.coreos.pool.ntp.org).
Aug 31 23:47:24 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: eth0: Updating address: 172.31.30.108/20 (valid for 1h)
Aug 31 23:47:24 ip-172-31-30-108.eu-central-1.compute.internal systemd-timesyncd[704]: Network configuration changed, trying to establish connection.
Aug 31 23:47:23 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCP CLIENT (0x5d6fbac2): T1 expires in 29min 58s
Aug 31 23:47:23 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCP CLIENT (0x5d6fbac2): T2 expires in 52min 28s
Aug 31 23:47:23 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCP CLIENT (0x5d6fbac2): lease expires in 59min 56s
Aug 31 23:47:22 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCP CLIENT (0x5d6fbac2): ACK
Aug 31 23:47:21 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCP CLIENT (0x5d6fbac2): REQUEST (renewing)
Aug 31 23:47:20 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:47:14,654][WARN ][monitor.jvm              ] [Jordan Seberius] [gc][young][66225][24540] duration [1s], collections [1]/[2s], total [1s]/[2.9m],
Aug 31 23:47:03 ip-172-31-30-108.eu-central-1.compute.internal docker[9743]: [2016-08-31 23:47:03,147][INFO ][monitor.jvm              ] [Jordan Seberius] [gc][young][66218][24539] duration [930ms], collections [1]/[2.1s], total [930ms]
Aug 31 23:46:00 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCPv6 CLIENT: Next retransmission in 2min 7s
Aug 31 23:46:00 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[1645]: DHCPv6 CLIENT: Sent SOLICIT
Aug 31 23:45:21 ip-172-31-30-108.eu-central-1.compute.internal locksmithd[809]: Unlocking old locks failed: [etcd.service etcd2.service] are inactive. Retrying in 5m0s.
Aug 31 23:45:21 ip-172-31-30-108.eu-central-1.compute.internal locksmithd[809]: [etcd.service etcd2.service] are inactive
```

### CoreOS Version ###

```
NAME=CoreOS
ID=coreos
VERSION=1068.10.0
VERSION_ID=1068.10.0
BUILD_ID=2016-08-23-0220
PRETTY_NAME="CoreOS 1068.10.0 (MoreOS)"
ANSI_COLOR="1;32"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://github.com/coreos/bugs/issues"
```

### Environment ###

EC2 instance (T2.small)

### Expected Behavior ###

The network connection shouldn't go down

### Actual Behavior ###

The network connection goes down after a few hours/days online

### Reproduction Steps ###

Provision the machine on EC2 with the cloud-config I provided and let it run for a few days

**Comment by styxlab** [comment]
According to the above logs your DHCP lease expired and could not be successfully renewed. It seems however, that problems started earlier than shown in the above logs. Could you please show the output of `journalctl -u systemd-networkd` for a timespan covering normal operation and including the network failures?

**Comment by canassa** [comment]
Thanks for you reply @styxlab,

According to our monitoring, the last time we lost connection with the server was on Sep 7 around 23:20 UTC, so I collected some logs from that point in time. There is a `-- Reboot --` on those logs at around Sep 8 08:13, that's a manual reboot that we did in order to bring the machine up again.

Here are the systemd-networkd logs:
https://gist.github.com/canassa/b35105f8fce21c5bbe317ea61c9a8558

I also included the other logs as they might be relevant:
https://gist.github.com/canassa/54d1dab25477864bebbd1041a1d949eb

Thank you!

**Comment by styxlab** [comment]
The important lines in the log file are:

`
Sep 07 22:41:52 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[657]: DHCP CLIENT (0xb87f2e70): STOPPED
Sep 07 22:41:45 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[657]: Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=Propert
Sep 07 22:41:08 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[657]: eth0: Failed
Sep 07 22:41:06 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[657]: eth0: Could not set DHCPv4 address: Connection timed out
Sep 07 22:39:02 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[657]: DHCP CLIENT (0xb87f2e70): T1 expires in 29min 44s
Sep 07 22:38:48 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[657]: DHCP CLIENT (0xb87f2e70): T2 expires in 52min 13s
Sep 07 22:38:36 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[657]: DHCP CLIENT (0xb87f2e70): lease expires in 59min 43s
Sep 07 22:38:18 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[657]: DHCP CLIENT (0xb87f2e70): ACK
Sep 07 22:37:53 ip-172-31-30-108.eu-central-1.compute.internal systemd-networkd[657]: DHCP CLIENT (0xb87f2e70): REQUEST (renewing)
`

So, a DHCP request for renewing was made, but it fails with setting the correct DHCPv4 address.

I had a similar issue and I had to tell the network-deamon explicitly which address space it should be requesting. Put the following in your cloud-config:

`
coreos:
  units:
    - name: 10-dhcp.network
      content: |
        [Match]
        Name=eth0
        [Network]
        DHCP=ipv4
        Address=172.31.30.108/20
        Gateway=172.31.1.1
        DNS=8.8.8.8
`

Please modify the above for the correct address, gateway and DNS.
Does this help?

**Comment by canassa** [comment]
Thanks, I updated my cloud-config with your suggestions and provisioned a new machine, I will monitor it to see how it goes.

side note:
A few days ago I also provisioned another CoreOS machine identical to the machine that's failing but this one I started without any cloud-config. This machine is running most almost one week now without any network errors so far. Therefore I have the suspicious that something on my cloud-config and/or docker containers might be helping the DHCP connection loss.
